packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m

```diff
@@ -0,0 +1,192 @@
+            @"code": @"unknown",
+            @"message": [error localizedDescription],
+        }];
+        return;
+      }
+
+      resolve([self getFirebaseVisionTextMap:text]);
+    }];
+  }];
+}
+
+RCT_EXPORT_METHOD(cloudTextRecognizerProcessImage:
+  (FIRApp *) firebaseApp
+    : (NSString *)filePath
+    : (NSDictionary *)cloudTextRecognizerOptions
+    : (RCTPromiseResolveBlock)resolve
+    : (RCTPromiseRejectBlock)reject
+) {
+  [RNFBMLVisionCommon UIImageForFilePath:filePath completion:^(NSArray *errorCodeMessageArray, UIImage *image) {
+    if (errorCodeMessageArray != nil) {
+      [RNFBSharedUtils rejectPromiseWithUserInfo:reject userInfo:(NSMutableDictionary *) @{
+          @"code": errorCodeMessageArray[0],
+          @"message": errorCodeMessageArray[1],
+      }];
+      return;
+    }
+
+    FIRVisionImage *visionImage = [[FIRVisionImage alloc] initWithImage:image];
+    FIRVision *vision = [FIRVision vision];
+
+    FIRVisionCloudTextRecognizerOptions *options = [[FIRVisionCloudTextRecognizerOptions alloc] init];
+
+    if (cloudTextRecognizerOptions[@"hintedLanguages"]) {
+      options.languageHints = cloudTextRecognizerOptions[@"hintedLanguages"];
+    }
+
+    NSInteger *modelType = [cloudTextRecognizerOptions[@"modelType"] pointerValue];
+    if (modelType == (NSInteger *) 1) {
+      options.modelType = FIRVisionCloudTextModelTypeSparse;
+    } else if (modelType == (NSInteger *) 2) {
+      options.modelType = FIRVisionCloudTextModelTypeDense;
+    }
+
+    FIRVisionTextRecognizer *textRecognizer = [vision cloudTextRecognizerWithOptions:options];
+
+    [textRecognizer processImage:visionImage completion:^(FIRVisionText *text, NSError *error) {
+      if (error != nil) {
+        [RNFBSharedUtils rejectPromiseWithUserInfo:reject userInfo:(NSMutableDictionary *) @{
+            @"code": @"unknown",
+            @"message": [error localizedDescription],
+        }];
+        return;
+      }
+
+      resolve([self getFirebaseVisionTextMap:text]);
+    }];
+  }];
+}
+
+- (NSDictionary *)getFirebaseVisionTextMap:(FIRVisionText *)text {
+  NSMutableDictionary *firebaseVisionTextMap = [[NSMutableDictionary alloc] init];
+
+  firebaseVisionTextMap[@"text"] = text.text;
+  firebaseVisionTextMap[@"blocks"] = [self getVisionTextBlocksList:text.blocks];
+
+  return firebaseVisionTextMap;
+}
+
+- (NSArray *)getVisionTextBlocksList:(NSArray <FIRVisionTextBlock *> *)blocks {
+  NSMutableArray *blockListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionTextBlock *block in blocks) {
+    NSMutableDictionary *textBlockFormatted = [[NSMutableDictionary alloc] init];
+
+    textBlockFormatted[@"text"] = block.text;
+    textBlockFormatted[@"confidence"] = block.confidence;
+    textBlockFormatted[@"recognizedLanguages"] = [self getLanguageCodesList:block.recognizedLanguages];
+//    textBlockFormatted[@"cornerPoints"] = [RNFBMLVisionCommon visionPointsToArray:block.cornerPoints];
```

Comment:
Corner points to point array issue

Reply:
Fixed in commit I just pushed from your Mac